### DISCUSSION TITLE
Different endpoint on different locale

### DISCUSSION THREAD

**Discussion by lapenji** [Diizzayy/nuxt-graphql-client#484] [Q&A] [answered]
Hello and thanks for your work!

I'm doing a nuxt ecommerce with shopify and I need to change the shopify graphql token and url based on my locale, is it possible? I'm defining my endpoints in nuxt.config.ts

thank you in advance :)

**Comment by Diizzayy** [comment, marked as answer]
Hey @lapenji, you should be able to leverage the [`useGqlHost`](https://nuxt-graphql-client.web.app/getting-started/composables#usegqlhost) and [`useGqlToken`](https://nuxt-graphql-client.web.app/getting-started/composables#usegqltoken) composables, to implement your logic based on the locale

**Comment by lapenji** [comment]
Thanks for your kind response, and your precious work of course :)

I tried to do so but I need to do it in a middleware and it seems that even if the function runs without errors the host remains the same!

this is my code in middleware/api.ts

```
 //this function get host and token from the env 
  const { host, token } = determineGqlHostAndTokenFromLocale(currentLocale); 
  useGqlHost(host, "shopify"); 
  useGqlToken(token, { 
    client: "shopify", 
    config: { 
      name: "X-Shopify-Storefront-Access-Token", 
      type: null, 
      value: token, 
    }, 
  }); 
```

if I log host the value is correct but if I look at the query results I find that it just queried the default endpoint.

I'm quite a noob so there is probably something I'm getting wrong! thanks again for your help!

**Reply by lapenji** [reply]
In the end I used graphql-request in my server side, I mark the previous answer as answer :) thank you again for your precious work!
